docs/usage/usage-guide.md

```diff
@@ -700,3 +700,208 @@ interface ThunkAPI {
+  name: 'users',
+  initialState: {
+    ids: [],
+    entities: {}
+  },
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(fetchUsers.fulfilled, (state, action) => {
+      // reduce the collection by the id property into a shape of { 1: { ...user }}
+      const byId = action.payload.users.reduce((byId, user) => {
+        byId[user.id] = user
+        return byId
+      }, {})
+      state.entities = byId
+      state.ids = Object.keys(byId)
+    })
+  }
+})
+```
+
+#### Normalizing with normalizr
+
+This is a very basic example of using the normalizr library with redux-toolkit
+
+```js
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
+import { normalize, schema } from 'normalizr'
+
+import userAPI from './userAPI'
+
+const userEntity = new schema.Entity('users')
+
+export const fetchUsers = createAsyncThunk('users/fetchAll', async () => {
+  const response = await userAPI.fetchAll()
+  return response.data
+})
+
+export const slice = createSlice({
+  name: 'users',
+  initialState: {
+    ids: [],
+    entities: {}
+  },
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(fetchUsers.fulfilled, (state, action) => {
+      const normalizedData = normalize(action.payload, userEntity)
+      state.entities = normalizedData.entities.users
+      state.ids = Object.keys(normalizedData.entities.users)
+    })
+  }
+})
+```
+
+#### Normalizing with `createEntityAdapter`
+
+This is an identitical implementation using RTK's `createEntityAdapter`.
+
+```js
+import {
+  createSlice,
+  createAsyncThunk,
+  createEntityAdapter
+} from '@reduxjs/toolkit'
+import userAPI from './userAPI'
+
+export const fetchUsers = createAsyncThunk('users/fetchAll', async () => {
+  const response = await userAPI.fetchAll()
+  return response.data
+})
+
+export const usersAdapter = createEntityAdapter()
+
+// By default, `createEntityAdapter` gives you `{ ids: [], entities: {} }`. If you want to track 'loading' or other keys, you would initialize them here like this: `getInitialState({ loading: false, activeRequestId: null })`
+const initialState = usersAdapter.getInitialState()
+
+export const slice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    removeUser: usersAdapter.removeOne
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchUsers.fulfilled, (state, action) => {
+      usersAdapter.upsertMany(state, action.payload)
+    })
+  }
+})
+
+const reducer = slice.reducer
+export default reducer
+
+export const { removeUser } = slice.actions
+```
+
+#### Using selectors with `createEntityAdapter`
+
+The entity adapter providers a selector factory that generates the most common selectors for you. Taking the example above, we can add selectors to our `usersSlice` like this:
+
+```js
+const {
+  selectIds,
+  selectEntities,
+  selectAll,
+  selectTotal
+} = usersAdapter.getSelectors(state => state.users)
+
+export const selectUserIds = state => selectIds(state)
```

Comment:
These extra exports are redundant - that's what these functions already do.

If you want to rename them to be more specific, do it while destructuring:

```js
export const {
  selectIds: selectUserIds
} = usersAdapter.getSelectors(state => state.users);
```